oci.jq

```diff
@@ -87,3 +87,52 @@ def sort_manifests:
 	sort_by(.platform | sort_split_platform)
 	| sort_attestations
 ;
+
+def validate_media_type:
+    if .mediaType != "application/vnd.oci.image.index.v1+json" and .mediaType then # TODO drop the second half of this validation: https://github.com/moby/buildkit/issues/4595
+        error("unsupported index mediaType: " + .mediaType)
+    else . end
+;
+
+def validate_index_type:
+    if .schemaVersion != 2 then
+        error("unsupported schemaVersion: " + .schemaVersion)
+    else . end
+    | validate_media_type
+;
+
+# jq -s 'include "oci"; validate_oci_basic' dir/oci-layout dir/index.json 
+def validate_oci_basic:
+    if length != 2 then
+        error("Unexpected oci-layout. Expecting 'oci-layout' and 'index.json'")
+    else . end
+    | .[0] |= (
+        if .imageLayoutVersion != "1.0.0" then
+            error("unsupported imageLayoutVersion: " + .imageLayoutVersion)
+        else . end
+    )
+    | .[1] |= (
+        . | validate_index_type
+        | if .manifests | length != 1 then
+            error("expected only one manifests entry, not " + (.manifests | length | tostring))
+        else . end
+
+        | .manifests[0] |= (
+            validate_media_type
+            # TODO validate .digest somehow (`crane validate`?) - would also be good to validate all descriptors recursively
+            | if .size < 0 then
+                error("invalid descriptor size: " + .size)
+            else . end
+        )
+    )
+;
```

Comment:
This function is this function optimized

https://github.com/docker-library/meta/blob/subset/.github/workflows/build.yml#L191-L241

The new shell function would be:

```sh
validate-oci-layout() {
  local dir="$1"
  jq -L.scripts -s '
    include "oci";
    validate_oci_basic | true
  ' "$dir/oci-layout" "$dir/index.json" || return "$?"
  local manifest
  manifest="$dir/blobs/$(jq -r '.manifests[0].digest | sub(":"; "/")' "$dir/index.json")" || return "$?"
  jq -L.scripts -s '
    include "oci";
    if length != 1 then
      error("unexpected image index document count: " + length)
    else .[0] end
    | validate_index_type

    # TODO more validation?
  ' "$manifest" || return "$?"
}
```